networkx/algorithms/bipartite/extendability.py

```diff
@@ -0,0 +1,135 @@
+""" Provides a function for computing the extendability of a
+graph which is undirected, simple, connected and bipartite. """
+
+
+import networkx as nx
+from networkx.utils import not_implemented_for
+
+
+__all__ = [
+    "find_extendability",
+]
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+def find_extendability(G):
+    """Computes the extendability of a graph.
+
+    Definition
+    Graph G is $k$-extendable if and only if G has a perfect matching and every
+    set of $k$ independent edges can extend to perfect matching.
+
+    EXTENDABILITY PROBLEM
+    Input: A graph G and a positive integer $k$.
+    Output: Is G $k$-extendable?
+
+    In general case the above problem is co-NP-complete([2]).
+    If graph G is bipartite, then it can be decided in polynomial time([1]).
+
+    The maximization version of the EXTENDABILITY PROBLEM asks to compute the
+    maximum $k$ for which G is $k$-extendable.
+
+    Let G be a simple, connected, undirected and bipartite graph with a perfect
+    matching M and bipartition (U,V). The residual graph of G, denoted by $G_M$,
+    is the graph obtained from G by directing the edges of M from V to U and the
+    edges that do not belong to M from U to V.
+
+    Lemma([1])
+    Let M be a perfect matching of G. G is k-extendable if and only if its residual
+    graph $G_M$ is strongly connected and there are $k$ vertex-disjoint directed
+    paths between every vertex of U and every vertex of V.
+
+    Parameters
+    ----------
+
+        G : NetworkX Graph
+
+    Returns
+    -------
+
+        extendability: int
+
+    Raises
+    ------
+
+         NetworkXError
+           If the graph G is not simple.
+           If the graph G is disconnected.
+           If the graph G is not bipartite.
+           If the graph G does not contain a perfect matching.
+           If the residual graph of G is not strongly connected.
+
+    Notes
+    -----
+
+         Time complexity: O($n^3$ $m^2$))
+
+    References
+    ----------
+
+         ..[1] "A polynomial algorithm for the extendability problem in bipartite graphs",
+           J. Lakhal, L. Litzler, Information Processing Letters, 1998.
+         ..[2] "The matching extension problem in general graphs is co-NP-complete",
+           Jan Hackfeld, Arie M. C. A. Koster, Springer Nature, 2018.
+
+    """
+    # Graph G must be simple
+    for edge in G.edges:
+        if edge[0] == edge[1]:
+            raise nx.NetworkXError("Graph G is not simple")
+    # Graph G must be connected
+    if not nx.is_connected(G):
+        raise nx.NetworkXError("Graph G is not connected")
```

Comment:
```suggestion
    if not nx.is_connected(G):
        raise nx.NetworkXError("Graph G is not connected")
```
This is an example of a comment that's not really necessary - the code itself makes it very clear what's happening here.